packages/headless/src/features/commerce/common/actions.ts

```diff
@@ -22,6 +26,7 @@ import {SortBy, SortCriterion} from '../sort/sort';
 
 export type StateNeededByQueryCommerceAPI = ConfigurationSection &
   ProductListingV2Section &
+  RecommendationV2Section &
```

Comment:
Instead of changing this `StateNeededByQueryCommerceAPI` type, we could extend it specifically for recs. Here's an example of what I mean with facet search:
https://github.com/coveo/ui-kit/blob/2f88117c593eab09c92fc07fa2331c5e49f9a192/packages/headless/src/features/commerce/facets/facet-search-set/commerce-facet-search-state.ts#L7-L9